LEAF_Request_Portal/sources/FormEditor.php

```diff
@@ -33,44 +33,45 @@ public function __construct($db, $login)
      * @param bool $overwriteExisting - If true, matching IDs will be overwritten
      * @return number
      */
-    public function addIndicator($package, $overwriteExisting = false)
-    {
-        $package['parentID'] = $package['parentID'] == '' ? null : $package['parentID'];
+    function addIndicator($package, $overwriteExisting = false) {
+    	$package['parentID'] = $package['parentID'] == '' ? null : $package['parentID'];
 
-        if (!$overwriteExisting)
+        if(!$overwriteExisting) 
         {
-            $vars = array(':name' => trim($package['name']),
-                ':format' => $package['format'],
-                ':description' => trim($package['description']),
-                ':default' => trim($package['default']),
-                ':parentID' => $package['parentID'],
-                ':categoryID' => $package['categoryID'],
-                ':html' => $package['html'],
-                ':htmlPrint' => $package['htmlPrint'],
-                ':required' => $package['required'],
-                ':sort' => isset($package['sort']) ? $package['sort'] : 1, );
-
-            $this->db->prepared_query('INSERT INTO indicators (indicatorID, name, format, description, `default`, parentID, categoryID, html, htmlPrint, required, sort, timeAdded, disabled)
-            								VALUES (null, :name, :format, :description, :default, :parentID, :categoryID, :html, :htmlPrint, :required, :sort, CURRENT_TIMESTAMP, 0)', $vars);
-        }
-        else
+    	    $vars = array(':name' => $package['name'],
```

Comment:
Are these variables getting sanitized?

Reply:
```$this->index['POST']->register('formEditor/newIndicator', function ($args) use ($formEditor) {
            $package = array();
            $package['name'] = XSSHelpers::sanitizeHTML($_POST['name']);
            $package['format'] = strip_tags($_POST['format']);
            $package['description'] = XSSHelpers::sanitizeHTML($_POST['description']);
            $package['default'] = XSSHelpers::sanitizeHTML($_POST['default']);
            $package['parentID'] = (int)$_POST['parentID'];
            $package['categoryID'] = XSSHelpers::xscrub($_POST['categoryID']);
            $package['html'] = $_POST['html'];
            $package['htmlPrint'] = $_POST['htmlPrint'];
            $package['required'] = (int)$_POST['required'];
            $package['is_sensitive'] = $_POST['is_sensitive'];
            $package['sort'] = (int)$_POST['sort'];

            return $formEditor->addIndicator($package);
        });```

In FormEditorController.php, this is the function's only usage.  The `'is_sensitive'`, `'html'`, and `'htmlPrint'` probably need sanitizing.

Reply:
Yeah, they all need to be.

Reply:
i don't think you need sanitize the html field